File: Irrgarten_Java/Irrgarten_P4/src/irrgarten/FuzzyPlayer.java

package irrgarten;

import java.util.ArrayList;

public class FuzzyPlayer extends Player{
    

    public FuzzyPlayer(Player other){
        super(other);
    }

    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        Directions preference=super.move(direction, validMoves);
        return Dice.nextStep(preference, validMoves, super.getIntelligence());
       
    }

    @Override
    public float attack(){
        return(Dice.intensity(super.getStrength())+super.sumWeapons());
    }

    @Override
    protected float defensiveEnergy(){
        return (Dice.intensity(super.getIntelligence())+super.sumShields());
    }

    @Override
    public String toString(){
        String cadena =  super.toString();
        return "Fuzzy " + cadena;
    }


}

File: Irrgarten_Ruby/Irrgarten_Ruby_P4/FuzzyPlayer.rb
#encoding:utf-8

require_relative 'Irrgarten'
require_relative 'Dice'
require_relative 'Weapon'
require_relative 'Shield'
require_relative 'Player'


module Irrgarten
	class FuzzyPlayer < Player
		def initialize()
			
		end
		
		public
		
		def copiar_player(otro)
			super(otro)
		end
		
		def move(direction, valid_moves)
			preference = super(direction,valid_moves)
			Dice.next_step(preference,valid_moves,@intelligence)
		end
		
		def attack()
			damage = Dice.intensity(@strength) + sum_weapons()
			return damage
		end
		
		def to_s()
			salida = super
			return "Fuzzy #{salida}"
		end
		
		protected
		
		def defensive_energy()
			defense = Dice.intensity(@intelligence) + sum_shields()
			return defense
		end
	end
end
